3Dmol/glviewer.js

```diff
@@ -2295,34 +2299,51 @@ $3Dmol.GLViewer = (function() {
             if (options.reps) {
                 reps = options.reps;
             }
-            var mostFrames = this.getFrames();
+            var mostFrames = this.getNumFrames();
             var that = this;
             var currFrame = 0;
             var inc = 1;
             var displayCount = 0;
             var displayMax = mostFrames * reps;
+            var time = new Date();
+            var resolve = function() {
+                that.render();
+                if (++displayCount == displayMax || !that.isAnimated()) {
+                    clearTimeout(intervalID);
+                    decAnim(); 
+                }
+                else {
+                    var newInterval = interval - (new Date().getSeconds() - time.getSeconds())/1000000.0;
```

Comment:
I was trying to mimic setTimeout to be the same setInterval. In setInterval, display would get called after certain interval regardless of how long it was taking to render whereas in recursive setTimeout, the interval would start getting counted only after the frame had been rendered. I did not want it to wait for some interval if the rendering time had already exceeded the interval.